src/unix/fs.c

```diff
@@ -1619,7 +1619,8 @@ static int uv__fs_statx(int fd,
     uv__store_relaxed(&no_statx, 1);
     return UV_ENOSYS;
   }
-
+  uv__msan_unpoison(&statxbuf, sizeof(struct uv__statx));
```

Comment:
I'd still want it inside the uv__statx - just check the error directly in there.